h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -552,16 +552,16 @@ private static int initalizeFromChunkIds(Frame f, int[] chunks, Vec.Reader[] vec
                 // enlarge final data arrays by 2x if needed
 
                 for (int j = 0; j < di._cats; ++j) {
-                    if (!vecs[j].isNA(i)) {
+                    if (vecs[j].at(i) != 0) {
```

Comment:
This seems incorrect. You are skipping values that have categorical level = 0. If categorical has level 2, you will correctly output (0, 0, 1), but if it has level 0 you will not output anything.

Reply:
Yes, this is incorrect.